db/db_iter.h

```diff
@@ -132,6 +132,7 @@ class DBIter final : public Iterator {
     ResetInternalKeysSkippedCounter();
     local_stats_.BumpGlobalStatistics(statistics_);
     iter_.DeleteIter(arena_mode_);
+    status_.PermitUncheckedError();
```

Comment:
If we call `status_.PermitUncheckedError()` here, we won't benefit from the enforcement of status check, right?

Reply:
Yes. But status_ is mostly used as a "carrier" of the status and return it to the user, when status() is called. So, it is reasonable to call `PermitUncheckedError()` in the destructor

Reply:
iiuc, if it's not returned to the user, adding `PermitUncheckedError()` will bypass the check. I am not sure if this is what we would like to achieve.

Reply:
> iiuc, if it's not returned to the user, adding `PermitUncheckedError()` will bypass the check. I am not sure if this is what we would like to achieve.

It is a little problematic. We have some classes that use Status or IOStatus as member. In most case, they are just used to carry the status through the life time of the class instance. And usually, user call a function to get the status when they want. In those cases, we are not able to know if user get the status or not.

Reply:
I agree with @riversand963 that, this thing will defeat the whole purpose of the status check for iterators.
I would rather fix those unit tests than do it here.

Reply:
If you want to remove the PermitUncheckedErrors in destructors, you need a means of telling if the Status was ever checked or set.  I see two potential solutions to this.  One is that an "empty Status" (e.g. Status s;) always passes the check.  Another is that Status::OK() always passes.  If choosing the latter, you would need to explicitly initialize the Status to OK and to change OK to be a static Status (rather than one created every time).

I think there are likely going to be places where an Iterator is created and the status is never set that are causing some of the assertions, and those are very hard to track down.